jumpscale/tools/syncer/__init__.py

```diff
@@ -0,0 +1,128 @@
+from jumpscale.god import j
+from watchdog.events import FileSystemEventHandler, PatternMatchingEventHandler
+from watchdog.observers import Observer
+import gevent
+
+DEFAULT_IGNORED_PATTERNS = [".git", ".pyc", '__pycache__', ".swp", '.swx']
+
+class Syncer(PatternMatchingEventHandler):
+
+    def __init__(self, sshclients_names, paths=None, patterns=None, ignore_patterns=None, ignore_directories=False, case_sensitive=False):
+        ignore_patterns = ignore_patterns or DEFAULT_IGNORED_PATTERNS
+        super().__init__(patterns, ignore_patterns, ignore_directories, case_sensitive)
+        self.observer = Observer()
+        self.sshclients_names = sshclients_names
+        self.paths = paths or {} # src:dst
+
+    def _get_dest_path(self, src_path):
+        print("paths: {} and path: {}".format(self.paths, src_path))
+
+        for path in self.paths.keys():
+            if path.startswith(src_path):
```

Comment:
Why using startswith here? Shouldn't it be equal? Especially since you are using `src_path` as a key

Reply:
paths: `{ '/tmp/myproj':'/root/myproj'}`
src_path would be a path prefixed by one of original paths e.g. `/tmp/myproj/myfile1`, so i need to check if the `src_path`  `startswith` one of the `paths.keys()`

Reply:
I saw what you meant definitely a typo